src/Esfa.Vacancy.Register.Api/Controllers/SearchApprenticeshipVacanciesController.cs

```diff
@@ -108,7 +108,7 @@ public SearchApprenticeshipVacanciesController(SearchApprenticeshipVacanciesOrch
         [SwaggerResponse(HttpStatusCode.BadRequest, "Failed request validation", typeof(BadRequestContent))]
         public async Task<IHttpActionResult> Search([FromUri(Name = "")]SearchApprenticeshipParameters searchApprenticeshipParameters)
         {
-            var results = await _searchOrchestrator.SearchApprenticeship(searchApprenticeshipParameters);
+            var results = await _searchOrchestrator.SearchApprenticeship(searchApprenticeshipParameters, Url);
```

Comment:
Although there is mapping being done in the orchestrator. For this part where the url resolving needs to happen, I would do setting of those url properties in the controller. We use the orchestrator to do the work that a controller action would do that doesn't rely on HTTPContext and so those parts are easier to test around without lots of HTTP mocking.

Reply:
Mapping routine has changed